pulsar-function-go/pf/context.go

```diff
@@ -153,8 +156,9 @@ func (c *FunctionContext) GetUserConfMap() map[string]interface{} {
 }
 
 // NewOutputMessage send message to the topic @param topicName: The name of the
-// topic for output message
+// topic for output message (DEPRECATED)
 func (c *FunctionContext) NewOutputMessage(topicName string) pulsar.Producer {
+	log.Warn("NewOutputMessage is deprecated, please use Publish() or PublishMsg()")
```

Comment:
We should add a new function like `NewProducer(topicName string) (pulsar.Producer, error)` or with any proper name. the reason is we need to remain this function exists, and let user know with the `log.Warn` if they still want this.

Reply:
Normally I would agree, but as I've [discussed](https://github.com/apache/pulsar/pull/9515#discussion_r578621233) [elsewhere](https://github.com/apache/pulsar/pull/9513#issuecomment-781516820), the pulsar function SDK API is relatively lightweight by design. The more knobs you give the user to tweak at runtime, the further you get from the control of the process the SDK appears to be designed for. At that point, it is not much different from a regular consumer/producer with deployment, some configuration and some logging taken care of.